### PR TITLE
Topic/ownership

### PR DESCRIPTION
**Pull request by danzelenak-usgs**

### CODE REVIEW

File: playbook/espa-worker/defaults/main.yml [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+# production info
 app_user: espa
 app_uid: 16802
 app_group: ie
 app_gid: 502
 
+# test info
+apt_user: espatst
+apt_uid: 16342
+apt_group: espatst
```
Comment (caustin-usgs) [inline playbook/espa-worker/defaults/main.yml]: i believe the apt_group and apd_group also need to be 'ie', according to my testing

File: playbook/espa-worker/tasks/user-group.yml [hunk 2]
```diff
@@ -14,3 +14,38 @@
     name: "{{ app_user }}"
     uid: "{{ app_uid }}"
     group: "{{ app_group }}"
+
+- name: user-group|Add espatst group
```
Comment (caustin-usgs) [inline playbook/espa-worker/tasks/user-group.yml]: per my other comment on the 'ie' group, do we need the espatst and espadev groups?

File: playbook/espa-worker/tasks/user-group.yml [hunk 3]
```diff
@@ -14,3 +14,38 @@
...
+    gid: "{{ apd_gid }}"
+    state: present
+
+- name: user-group|Add espadev user
+  user:
+    name: "{{ apd_user }}"
+    uid: "{{ apd_uid }}"
+    group: "{{ apd_group }}"
+
+- name: install sudo
```
Comment (caustin-usgs) [inline playbook/espa-worker/tasks/user-group.yml]: can skip this with the 'ie' group

File: processing/config.py [hunk 4]
```diff
@@ -56,6 +56,7 @@ def config():
         de('espa_priority', None),
         de('espa_schema', '/usr/local/schema/espa_internal_metadata_v2_1.xsd'),
         de('espa_user', None),
+        de('espa_group', None),
```
Comment (caustin-usgs) [inline processing/config.py]: this will require adding espa_group to the scheduler setup too

File: processing/distribution.py [hunk 5]
```diff
@@ -18,13 +18,14 @@
...
 from environment import Environment, DISTRIBUTION_METHOD_LOCAL
+from utilities import change_ownership
 from espa_exception import ESPAException
 import sensor
 import transfer
 
 
 def package_product(immutability, source_directory, destination_directory,
-                    product_name):
+                    product_name, user, group):
```
Comment (caustin-usgs) [inline processing/distribution.py]: are the new user or group args getting used in this function?

File: processing/distribution.py [hunk 6]
```diff
@@ -616,7 +618,9 @@ def distribute_product_local(immutability, product_name, source_path,
                      local_cksum_value) = package_product(immutability,
                                                           source_path,
                                                           packaging_path,
-                                                          product_name)
+                                                          product_name,
+                                                          user,
```
Comment (caustin-usgs) [inline processing/distribution.py]: it doesn't look like user and group get used in the package_product function

File: processing/distribution.py [hunk 7]
```diff
@@ -773,7 +784,9 @@ def distribute_product(immutability, product_name, source_path,
             distribute_product_local(immutability,
                                      product_name,
                                      source_path,
-                                     package_path)
+                                     package_path,
+                                     user,
+                                     group)
 
     else:  # remote
```
Comment (caustin-usgs) [inline processing/distribution.py]: like you mentioned yesterday, after this, we should strip out the unused bits of this codebase. the 'remote' options, etc

File: processing/logging_tools.py [hunk 8]
```diff
@@ -329,18 +329,20 @@ def archive_log_files(order_id, product_id):
...
-        shutil.copyfile(full_logfile_path, destination_file)
+        shutil.copyfile(full_logfile_path, destination_job_file)
 
         # Mapper log file
         full_logfile_path = os.path.abspath(WORKER_LOG_FILENAME)
         final_log_name = '-'.join([WORKER_LOG_PREFIX, order_id, product_id])
         final_log_name = '.'.join([final_log_name, 'log'])
         # Determine full destination
-        destination_file = os.path.join(destination_path, final_log_name)
+        destination_mapper_file = os.path.join(destination_path, final_log_name)
```
Comment (caustin-usgs) [inline processing/logging_tools.py]: 👍

File: processing/logging_tools.py [hunk 9]
```diff
@@ -329,18 +329,20 @@ def archive_log_files(order_id, product_id):
...
         # Copy it
-        shutil.copyfile(full_logfile_path, destination_file)
+        shutil.copyfile(full_logfile_path, destination_mapper_file)
+
+        return destination_path
```
Comment (caustin-usgs) [inline processing/logging_tools.py]: 👍

File: processing/config.py (hunk 4 above)
Comment (danzelenak-usgs) [reply processing/config.py]: This is done - opened a PR on espa-scheduler for merging topic/vars

File: processing/distribution.py (hunk 5 above)
Comment (danzelenak-usgs) [reply processing/distribution.py]: Nope - these should have been taken out, leftovers from testing..

**Pull request merged**